src/NCronJob/Registry/JobDefinition.cs

```diff
@@ -22,22 +21,20 @@ private JobDefinition(
         string dynamicHash,
         JobExecutionAttributes jobPolicyMetadata)
     {
-        Type = typeof(DynamicJobFactory);
         IsTypedJob = false;
-        JobFullName = $"{typeof(DynamicJobFactory).Namespace}.{dynamicHash}";
-        JobName = dynamicHash;
+        JobFullName = $"Untyped job {typeof(DynamicJobFactory).Namespace}.{dynamicHash}";
 
         JobPolicyMetadata = jobPolicyMetadata;
     }
 
-    public Type Type { get; }
+    public string Name => CustomName is not null ? $"{CustomName} ({JobFullName})" : JobFullName;
```

Comment:
`Name` is used for all logging.

Reply:
Maybe renaming it to LoggingIdentifier would be a better fit?

Reply:
Name is fine - as it is exposed as can be used by any means of the user.